### ISSUE TITLE
[TypeScript] index.d.ts not copied under `node_modules/react-json-view/`

### ISSUE THREAD

**Issue by franklixuefei** [mac-s-g/react-json-view#188]
After pulling the latest version (1.16.2), I still don't see `index.d.ts` under `node_modules/react-json-view/`, nor is it under `dist`. Is there any steps missing? Could you look into this?

![image](https://user-images.githubusercontent.com/1334982/39439348-22e5d2ba-4c5c-11e8-97d4-f6bce319d846.png)

![image](https://user-images.githubusercontent.com/1334982/39439359-29a08550-4c5c-11e8-9a68-18cad0008d2f.png)

**Comment by mac-s-g** [comment]
Sorry, I'm not using typescript, so I'm doing this somewhat blind.

I merged the changes and ran the build.  Does the build need to produce the typescript definitions as an additional asset?

You can test the build output by pulling the code and running `npm run build`.

Make sure you see what you need reflected in your local `/dist/` folder after running the build.

**Comment by franklixuefei** [comment]
@mac-s-g I looked into this, and it seems that it is unrelated to the result of running `npm run build`. `npm run build` builds the code into `main.js`, which is all it is expected to do. However, `index.d.ts` should bu published together with other files like `LICENSE`, `README.md`, etc, outside of `dist` folder, directly under `react-json-view` folder. This is related to how you publish the package. I believe you are using `npm publish` to publish the package, and it will look for `.npmignore` to exclude what you don't want to be published. However, I also looked at that file, and `index.d.ts` does not match any pattern in it, so it should have been published along with other files, and I should have seen it in my `node_modules`. Could you double check if you included `index.d.ts` when you publish your package?

**Comment by mac-s-g** [comment]
okay, looking into it now

**Comment by mac-s-g** [comment]
added this: https://github.com/mac-s-g/react-json-view/commit/a22907ccb40e132698382ae0edcb61a29add4f81

**Comment by mac-s-g** [comment]
and published to [1.16.3](https://www.npmjs.com/package/react-json-view)

**Comment by franklixuefei** [comment]
@mac-s-g Sorry just got a chance to reply. Unfortunately, in my project, I still see this.
![image](https://user-images.githubusercontent.com/1334982/39455330-b05ea670-4c94-11e8-81e6-effa66b8f355.png)

However, if I manually move `index.d.ts` out of `dist`, it works. like this
![image](https://user-images.githubusercontent.com/1334982/39455338-bc7f6e62-4c94-11e8-9ac9-77a5d72e5569.png)

**Comment by franklixuefei** [comment]
@mac-s-g Sorry for spamming, but I asked someone who also publishes typescript definition file for their project. He just uses `npm publish` and leverages `npmignore` to ignore the files he doesn't want to publish, which is what you are doing as well. I think this is it, and `index.d.ts` should appear in the published package as long as it is not ignored. Not sure if this helps you.

FYI - https://github.com/pburtchaell/redux-promise-middleware/issues/211

**Comment by mac-s-g** [comment]
Thanks, I think I see what needs to be done. I'll give it a shot when I get
home tonight.

**Comment by franklixuefei** [comment]
@mac-s-g BTW, this line can be reverted once you have the right fix.

https://github.com/mac-s-g/react-json-view/blob/a22907ccb40e132698382ae0edcb61a29add4f81/package.json#L58

**Comment by mac-s-g** [comment]
yea, i've changed that back locally.  I'm not really sure why `index.d.ts` didn't get published with 1.16.2.  The file does not exist in `.npmignore`.

**Comment by mac-s-g** [comment]
i think i figured it out.  gimme a sec.

**Comment by mac-s-g** [comment]
the deed is done! :raised_hands: 
https://unpkg.com/react-json-view@1.16.5/

the `files` key in `package.json` was gating my published files to only include `/dist/` (and a few other whitelisted files).  Now `.npmignore` should be the only files missing from the published module.

thanks for working through that with me @franklixuefei .  let me know if the 1.16.5 is working for you!

**Comment by franklixuefei** [comment]
@mac-s-g Thanks! Looks great! This is exactly what I was looking for. Thanks a lot for working this out with me!

**Comment by garygentry** [comment]
../../node_modules/react-json-view/index.d.ts(268,1): error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.

const ReactJson: React.ComponentType<ReactJsonViewProps>;
    ~~~~~

-----------------

line 267 should be:

declare const ReactJson: React.ComponentType<ReactJsonViewProps>;

**Comment by franklixuefei** [comment]
@garygentry Thanks for reporting! Hmm weird, it doesn't repro for me... Which typescript version are you using? I'm using 2.8.1. Probably that matters. I tried both with and without `declare`, and it turned out that both work for me. I'll file a PR for adding `declare` to line 268.

**Comment by kompot** [comment]
@franklixuefei I tried both 2.8.1 and 2.8.3 - error mentioned by @garygentry is in both for me (and your fix obviously fixes that, waiting for PR to be merged, thx)

**Comment by garygentry** [comment]
Originally 2.7.2, but just tried 2.8.3 and same thing.  This breaks the entire build so looking forward to merge..

**Comment by franklixuefei** [comment]
@garygentry @kompot Sorry about this guys. Until this PR is merged, you can roll back to using v1.16.1.

**Comment by mac-s-g** [comment]
hopping on this now

**Comment by mac-s-g** [comment]
fix is merged and published under 1.16.6

**Comment by franklixuefei** [comment]
@mac-s-g Thanks Mac!
